src/main/java/com/itechart/javalab/library/dao/impl/SqlBookDao.java

```diff
@@ -23,4 +37,50 @@ public static BookDao getInstance() {
         return instance;
     }
 
+
+    @Override
+    public Optional<List<Book>> getBooks() {
+
+        List<Book> books;
+
+        try (Connection connection = connectionPool.getConnection()) {
+            try (PreparedStatement preparedStatement = connection.prepareStatement(GET_ALL_BOOKS)) {
+                ResultSet resultSet = preparedStatement.executeQuery();
+                Map<Integer, Book> tempBooks = new HashMap<>();
+
+                while (resultSet.next()) {
+                    int id = resultSet.getInt("book.id");
+                    if (tempBooks.containsKey(id)) {
+                        Book existsBook = tempBooks.get(id);
+                        existsBook.getAuthor().add(buildAuthor(resultSet));
+                    } else {
+                        Book book = buildBook(resultSet);
+                        book.setId(id);
+                        tempBooks.put(id, book);
+                    }
+                }
+                books = new ArrayList<>((tempBooks.values()));
+            }
+        } catch (SQLException e) {
+            log.error("SqlException in attempt to get Connection", e);
+            return Optional.empty();
+        }
+        return Optional.of(books);
+    }
+
+    private Author buildAuthor(ResultSet resultSet) throws SQLException {
+        int authorId = resultSet.getInt("author.id");
+        String authorName = resultSet.getString("name");
+        return Author.builder().id(authorId).name(authorName).build();
+    }
+
+    private Book buildBook(ResultSet resultSet) throws SQLException {
+        String title = resultSet.getString("title");
+        LocalDateTime publishDate = resultSet.getTimestamp("publish date").toLocalDateTime();
+        int inStock = resultSet.getInt("in_stock");
+        Set<Author> authors = new HashSet<>();
+        authors.add(buildAuthor(resultSet));
+        return Book.builder().title(title).publishDate(publishDate).inStock(inStock).author(authors).build();
+    }
```

Comment:
`Book` and the `Author` must be responsible for their own creation. `Book.createFrom(resultSet)`